### PR TITLE
fix(core): stop to add layer if there's a error

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
Stop to add layer if there's a function error

## Motivation and Context
The layers work is disrupted
By example, if a layer is added with the same id, it's still add.
if we delete a layer then both are deleted

### CODE REVIEW

**Review by zarov — APPROVED** [review]
LGTM, good catch thanks

File: src/Core/Prefab/GlobeView.js [hunk 1]
```diff
@@ -281,6 +281,9 @@ GlobeView.prototype = Object.create(View.prototype);
 GlobeView.prototype.constructor = GlobeView;
 
 GlobeView.prototype.addLayer = function addLayer(layer) {
+    if (!layer) {
```
Comment (peppsac) [inline src/Core/Prefab/GlobeView.js:283]: Are these check really necessary ?
(if they are, you should add them to View.addLayer and PanoramaView.addLayer as well)

Comment (autra) [reply src/Core/Prefab/GlobeView.js:283]: Also, this breaks current code: now this function can return undefined (it couldn't before). Please consider returning a rejected promise with a suitable `Error` object.

File: src/Core/View.js [hunk 2]
```diff
@@ -325,16 +326,19 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
...
         if (parentLayer) {
             parentLayer.attach(layer);
         } else {
             if (typeof (layer.update) !== 'function') {
                 reject(new Error('Cant add GeometryLayer: missing a update function'));
+                return;
             }
             if (typeof (layer.preUpdate) !== 'function') {
                 reject(new Error('Cant add GeometryLayer: missing a preUpdate function'));
+                return;
```
Comment (autra) [inline src/Core/View.js:351]: :+1:

File: src/Core/Prefab/GlobeView.js (hunk 1 above)
Comment (gchoqueux) [reply src/Core/Prefab/GlobeView.js:283]: yes it is necessary because this error is not handle
I added error handling in all `addLayer`

Comment (autra) [reply src/Core/Prefab/GlobeView.js:283]: At the same time, is it really necessary to do so? If we let the code like it is before, users would have a rejected promise with "Error cannot read "type" of undefined" or something like that, which is imho clear enough. Anyway, why not having a better error message.

File: src/Core/Prefab/GlobeView.js [hunk 3]
```diff
@@ -280,6 +280,9 @@ GlobeView.prototype = Object.create(View.prototype);
...
+        return new Promise((resolve, reject) => reject(new Error('layer is undefined')));
```
Comment (autra) [inline src/Core/Prefab/GlobeView.js:284]: (you can do 

```js
return Promise.reject(new Error(...))
```
btw )

File: src/Core/View.js [hunk 4]
```diff
@@ -319,9 +319,14 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
         layer.update = layer.update || updateLayeredMaterialNodeElevation;
     }
     return new Promise((resolve, reject) => {
+        if (!layer) {
```
Comment (autra) [inline src/Core/View.js:322]: Oh we missed that: this is dead code, if layer is undefined, it'd throw on line 316 before reaching this.

**Review by autra — COMMENTED** [review]
2 nits, if someone wants to make a followup.

**Pull request merged**